### PR TITLE
ACL bootstrap test

### PR DESCRIPTION
**Pull request by chelseakomlo**

### CODE REVIEW

**Comment by chelseakomlo** [comment]
Note that how these tests are run will be improved on another PR. Currently tests are run within a docker container, but in the future the cluster will run within a docker container but the test will run on the host. This is keep each test isolated to prevent flakiness.

File: e2e/migrations/docker-run.sh [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 
-docker run --privileged -v \
+testOne= docker run --privileged -v \
 $ROOT_DIRECTORY:/gopkg/src/github.com/hashicorp/nomad \
 -it nomad-e2e /bin/bash \
--c "cd gopkg/src/github.com/hashicorp/nomad/e2e/migrations && go test -integration"
+-c "cd gopkg/src/github.com/hashicorp/nomad/e2e/migrations && go test --run \
+TestJobMigrations -integration"
+
+echo $testOne
```
Comment (dadgar) [inline e2e/migrations/docker-run.sh]: Are you not running test one anymore?

File: e2e/migrations/migrations_test.go [hunk 2]
```diff
@@ -108,15 +109,26 @@ func isSuccess(execCmd *exec.Cmd, retries int, keyword string) (string, error) {
 
 // allNodesAreReady attempts to query the status of a cluster a specific number
 // of times
-func allNodesAreReady(retries int) (string, error) {
-	cmd := exec.Command("nomad", "node-status")
+func allNodesAreReady(retries int, flags string) (string, error) {
```
Comment (dadgar) [inline e2e/migrations/migrations_test.go]: I'm not a huge fan of these. They seem super fragile but I couldn't see a better way with the commands we have. This is a sign that we need to improve them

Comment (chelseakomlo) [reply e2e/migrations/migrations_test.go]: I completely agree. Alternatively, I could do these checks via the HTTP API, which would be less fragile but would not be a complete e2e test. Open to any other suggestions.

File: e2e/migrations/docker-run.sh (hunk 1 above)
Comment (chelseakomlo) [reply e2e/migrations/docker-run.sh]: This was just for printing test output, but is unnecessary. Removed.

**Pull request merged**